meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -0,0 +1,213 @@
+        new EnumMap<>(org.apache.helix.metaclient.api.Op.Type.class);
+  }
+
+  private static Map<org.apache.helix.metaclient.api.Op.Type, Function<org.apache.helix.metaclient.api.Op, Op>> getOpMap() {
+    return OpMapHolder.OPMAP;
+  }
+
+  private static void initializeOpMap() {
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CREATE, op -> {
+      try {
+        CreateMode mode = convertMetaClientMode(((org.apache.helix.metaclient.api.Op.Create) op).getEntryMode());
+        return Op.create(op.getPath(), ((org.apache.helix.metaclient.api.Op.Create) op).getData(), DEFAULT_ACL, mode);
+      } catch (KeeperException e) {
+        throw translateZkExceptionToMetaclientException(ZkException.create(e));
+      }
+    });
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.DELETE,
+        op -> Op.delete(op.getPath(), ((org.apache.helix.metaclient.api.Op.Delete) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.SET,
+        op -> Op.setData(op.getPath(),
+            ((org.apache.helix.metaclient.api.Op.Set) op).getData(), ((org.apache.helix.metaclient.api.Op.Set) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CHECK,
+        op -> Op.check(op.getPath(), ((org.apache.helix.metaclient.api.Op.Check) op).getVersion()));
+  }
+
+  private static CreateMode convertMetaClientMode(MetaClientInterface.EntryMode entryMode) throws KeeperException {
+    switch (entryMode) {
+      case PERSISTENT:
+        return CreateMode.PERSISTENT;
+      case EPHEMERAL:
+        return CreateMode.EPHEMERAL;
+      case CONTAINER:
+        return CreateMode.CONTAINER;
+      default:
+        throw new IllegalArgumentException(entryMode.name() + " is not a supported EntryMode.");
+    }
+  }
+
+  /**
+   * Helper function for transactionOP. Converts the result from calling zk transactional support into
+   * metaclient OpResults.
+   * @param zkResult
+   * @return
+   */
+  public static List<OpResult> zkOpResultToMetaClientOpResults(List<org.apache.zookeeper.OpResult> zkResult) {
+    if (getOpResultMap().isEmpty()) {
+      initializeOpResultMap();
+    }
+    List<OpResult> metaClientOpResult = new ArrayList<>();
+    for (org.apache.zookeeper.OpResult opResult : zkResult) {
+      Function<org.apache.zookeeper.OpResult, OpResult> function = getOpResultMap().get(opResult.getClass());
+      if (function != null) {
+        metaClientOpResult.add(function.apply(opResult));
+      } else {
+        throw new IllegalArgumentException("OpResult type " + opResult.getType() + "is not supported.");
+      }
+    }
+
+    return metaClientOpResult;
+  }
+
+  private static final class OpResultMapHolder {
+    static final Map<Class<? extends org.apache.zookeeper.OpResult>, Function<org.apache.zookeeper.OpResult, OpResult>> OPRESULTMAP =
+        new HashMap<>();
+  }
```

Comment:
Thanks for the work Marcos. 
I see the current lazy loading functionally works, but it's kind of redundant with this holder. You might miss the point of using a on-demand holder. I believe we can move the init() method inline. Please refer to JVM classloading and on-demand holder idiom.

Reply:
I removed the unnecessary null check before each "op translate" as initializing the map guarantees non null value of opMap and opResultMap. Let me know if that addresses the comment, thanks!

Reply:
Has this been addressed? Why initializeOpResultMap is not invoked for OPRESULTMAP?

Reply:
Correct me if I am wrong, but I believe what you're saying is to have the map be populated when the static variable `OPRESULTMAP` is declared? I have attempted several methods and none seemed feasible:
1. Create a constructor that invokes `initializeOpResultMap` method when `OpResultMapHolder` object is created. However, given the nature of it being a util class, making it an object is counterproductive.
2. Populating the map directly when declaring the `OPRESULTMAP` static variable instead of having a separate method. This can only be done with java JDK 9+ (current library is at 8) so decided against it.

With the current implementation, `initializeOpResultMap` is called just once when needed in the `zkOpResultToMetaClientOpResults` method. Inside, `getOpResultMap` is called first which initializes the `OPRESULTMAP` for the first time, effectively achieving lazy loading. Though ideally when initializing `OPRESULTMAP` for the first time, we would like it to automatically be populated with its values, as mentioned above, I couldn't find a reasonable way to do so. Please suggest a solution that I may have missed.

Reply:
This is incorrect. In your impl, `initializeOpResultMap` is called every time `zkOpResultToMetaClientOpResults` is called, which trace back to `ZkMetaClient.transactionOP`. 
I don't get why point 2 is not applicable. To achieve lazy loading, you can combine the "initialize" and "value population" in your static method `initializeOpResultMap` by changing the signature and content to build and return the map, so you can call it directly at declaration.

Reply:
Thank you for your correction, I understand what you mean. I was implementing option 2 wrong which was why the error occurred. Thanks for the feedback and I have implemented those changes, let me know if it's good!